src/utils/upload/index.ts

```diff
@@ -120,3 +131,60 @@ export function areFileWithCollectionsSame(
 ): boolean {
     return firstFile.localID === secondFile.localID;
 }
+
+export function analyseUploadFiles(
+    uploadType: UPLOAD_TYPE,
+    toUploadFiles: File[] | ElectronFile[]
+): AnalysisResult {
+    if (isElectron() && uploadType === UPLOAD_TYPE.FILES) {
+        return NULL_ANALYSIS_RESULT;
+    }
+
+    const paths: string[] = toUploadFiles.map((file) => file['path']);
+    const getCharCount = (str: string) => (str.match(/\//g) ?? []).length;
+    paths.sort((path1, path2) => getCharCount(path1) - getCharCount(path2));
+    const firstPath = paths[0];
+    const lastPath = paths[paths.length - 1];
+
+    const L = firstPath.length;
+    let i = 0;
+    const firstFileFolder = firstPath.substring(0, firstPath.lastIndexOf('/'));
+    const lastFileFolder = lastPath.substring(0, lastPath.lastIndexOf('/'));
+    while (i < L && firstPath.charAt(i) === lastPath.charAt(i)) i++;
+    let commonPathPrefix = firstPath.substring(0, i);
+
+    if (commonPathPrefix) {
+        commonPathPrefix = commonPathPrefix.substring(
+            0,
+            commonPathPrefix.lastIndexOf('/')
+        );
+        if (commonPathPrefix) {
+            commonPathPrefix = commonPathPrefix.substring(
+                commonPathPrefix.lastIndexOf('/') + 1
+            );
+        }
+    }
+    return {
+        suggestedCollectionName: commonPathPrefix || null,
+        multipleFolders: firstFileFolder !== lastFileFolder,
+    };
+}
+export function getCollectionWiseFiles(toUploadFiles: File[] | ElectronFile[]) {
+    const collectionWiseFiles = new Map<string, (File | ElectronFile)[]>();
+    for (const file of toUploadFiles) {
+        const filePath = file['path'] as string;
```

Comment:
file['path'] will always be non-null/empty? Even when user has dragged few files right after sign up?

Reply:
yes,it will always be `not-null/empty` this is the path will respect to the root folder the user is trying to upload.

If user is just uploading bunch of files with then it will be the fileName